libsolidity/interface/StandardCompiler.cpp

```diff
@@ -463,9 +463,15 @@ std::optional<Json::Value> checkOptimizerDetailSteps(Json::Value const& _details
 	{
 		if (_details[_name].isString())
 		{
-			std::string const fullSequence = _details[_name].asString();
+			std::string fullSequence = _details[_name].asString();
+			boost::trim(fullSequence);
```

Comment:
You're trimming the string here (removing whitespace), but not doing so in the command line parser, which is why your check for an empty sequence string is failing (when supplying `"  "`). You can trim the string in `OptimiserSuite::isEmptyOptimizerSequence(std::string)` and remove the whitespace check from it.

Reply:
I agree with moving the trim inside `OptimiserSuite::isEmptyOptimizerSequence(std::string)`.
However, I think that the `empty()` check is needed still. For example, if the input is `u:`, it is not an empty optimizer sequence and so it should raise the error with the first part of the message, but not with the hint about the empty/blank string.

Reply:
> I agree with moving the trim inside `OptimiserSuite::isEmptyOptimizerSequence(std::string)`.

Although now I am not sure of that... since I think that we need to check for the empty and blank spaces case. For this latter case, I used the trim before checking for the empty string.

Reply:
By the way, note that the corresponding change in `CommandLineParser` does *not* modify metadata because there you use a fresh variable for the string that gets trimmed.

This inconsistency should have come up in tests so apparently we don't have any covering this. I think we should have a test showing how a sequence with whitespace looks when stored in metadata. Actually two, one for for CLI and one for Standard JSON.

Reply:
> It will actually affect metadata if you do it here.

Yes. I had not noticed that `fullSequence` becomes part of the metadata.






> This inconsistency should have come up in tests so apparently we don't have any covering this. I think we should have a test showing how a sequence with whitespace looks when stored in metadata. Actually two, one for for CLI and one for Standard JSON.

I will create an issue for that.

Reply:
I mean, I'd keep is simple so I'm not sure if creating an issue would be helpful. I'd just check if, as a part of this PR, you can fit a simple test for this into whatever test suites we already have. If there's no good place for it, It's fine to skip it. It would be helpful here but it's also not *that* important.

Reply:
Ok, I'll try that.

Reply:
I added some new tests and extend a few existing ones.